### PR TITLE
Duplications deleted

### PR DESCRIPTION
**Pull request by GhDj**
- [x] Bump PayPlug version with a 4th digit (ex: 1.1.0 -> 1.1.0.1)
- [x] Generate payplug_woocommerce.zip
- [x] Install payplug_woocommerce.zip on your Woocommerce environment
- [x] Login to your newly installed PayPlug plugin
- [x] Validate a simple payment with PayPlug
- [x] Validate a refund partial/total with PayPlug
- [x] Check apache logs

### CODE REVIEW

File: src/Gateway/PayplugResponse.php [hunk 1]
```diff
@@ -64,51 +57,131 @@ public function process_payment( $resource, $is_payment_with_token = false ) {
 		$metadata = PayplugWoocommerceHelper::extract_transaction_metadata( $resource );
 		$order_metadata = $order->get_meta('_payplug_metadata', true);
```
Comment (PPmmesquita) [inline src/Gateway/PayplugResponse.php:58]: $order_metadata - not used variable

**Review by PPmmesquita — CHANGES_REQUESTED** [review]

File: src/Gateway/PayplugResponse.php [hunk 2]
```diff
@@ -64,51 +57,131 @@ public function process_payment( $resource, $is_payment_with_token = false ) {
...
 
-		if ( ! empty( $resource->failure ) ) {
-			PayplugWoocommerceHelper::set_flag_ipn_order($order, $metadata, true);
-			$order->update_status(
-				'failed',
-				sprintf( __( 'PayPlug IPN OK | Transaction %s failed : %s', 'payplug' ), $resource->id, wc_clean( $resource->failure->message ) )
-			);
+		if ( isset( $resource->payment_method ) && is_array( $resource->payment_method ) ) {
+			if ( in_array( $resource->payment_method['type'], array( 'oney_x3_with_fees', 'oney_x4_with_fees' ) ) ) {
```
[the remark this answers is not in the document]
Comment (GhDj) [reply src/Gateway/PayplugResponse.php:61]: Yes, good point thank you

File: src/Gateway/PayplugResponse.php [hunk 3]
```diff
@@ -117,36 +190,34 @@ public function process_payment( $resource, $is_payment_with_token = false ) {
...
 
-	private function oney_ipn( $resource ) {
+	public function oney_ipn( $resource ) {
 
 		$order_id = wc_clean( $resource->metadata['order_id'] );
 		$order    = wc_get_order( $order_id );
 
-		$metadata = PayplugWoocommerceHelper::extract_transaction_metadata( $resource );
-		$order_metadata = $order->get_meta('_payplug_metadata', true);
+		$metadata       = PayplugWoocommerceHelper::extract_transaction_metadata( $resource );
```
[the remark this answers is not in the document]
Comment (GhDj) [reply src/Gateway/PayplugResponse.php:198]: true I just checked that, also $order_metadata on line 58 will be removed

**Comment by issamhamed2020** [comment]
array( 'oney_x3_with_fees', 'oney_x4_with_fees' )

should have also “without fees”:

array( 'oney_x3_with_fees', 'oney_x4_with_fees', 'oney_x3_without_fees', 'oney_x4_without_fees' )

**Comment by issam-payplug** [comment]
also for every check the same block of code is repeated (except $this->oney_ipn( $resource ); in the first check and $this->payplug_ipn($resource); in the second) so maybe this code can be put in a seperate function.
maybe we could let this to the refactoring later as we discussed before.

**Review by issam-payplug — CHANGES_REQUESTED** [review]
`array( 'oney_x3_with_fees', 'oney_x4_with_fees' )
`
should have also “without fees”:

`array( 'oney_x3_with_fees', 'oney_x4_with_fees', 'oney_x3_without_fees', 'oney_x4_without_fees' )`

File: src/Gateway/PayplugResponse.php [hunk 4]
```diff
@@ -55,14 +55,12 @@ public function process_payment( $resource, $is_payment_with_token = false ) {
 		}
 
 		$metadata = PayplugWoocommerceHelper::extract_transaction_metadata( $resource );
-		$order_metadata = $order->get_meta('_payplug_metadata', true);
 
 		if ( isset( $resource->payment_method ) && is_array( $resource->payment_method ) ) {
-			if ( in_array( $resource->payment_method['type'], array( 'oney_x3_with_fees', 'oney_x4_with_fees' ) ) ) {
+			if ( in_array( $resource->payment_method['type'], array( 'oney_x3_with_fees', 'oney_x4_with_fees',  'oney_x4_with_fees', 'oney_x3_without_fees', 'oney_x4_without_fees' ) ) ) {
```
Comment (PPmmesquita) [inline src/Gateway/PayplugResponse.php:60]: you've "oney_x4_with_fees" twice in the array

**Review by PPmmesquita — CHANGES_REQUESTED** [review]

**Review by PPmmesquita — APPROVED** [review]

**Review by issam-payplug — APPROVED** [review]

**Pull request merged**